packages/vite/src/node/plugins/resolve.ts

```diff
@@ -283,7 +278,7 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
 
         if (
           targetWeb &&
-          options.browserField &&
+          options.mainFields.includes('browser') &&
```

Comment:
Maybe it's better to assign this to a variable so that we don't need to loop over the `mainFields` many times.

Reply:
I just tried it locally and it's causing the code to drill everywhere that I'm not sure if it's worth it 😅 Some resolve options while a lot of parameters are mutually exclusive, but this extra variable would be deriving from the existing `resolveOptions` parameter, so I think maybe we can improve it later if we want to squeeze it.

Reply:
Ah, I see. I think it's fine in that case. 👍